src/components/DpIcon/DpIcon.vue

```diff
@@ -1,57 +1,90 @@
 <template>
-  <svg
-    class="c-icon"
-    :class="proportionClass"
-    :height="pxSize"
-    :width="pxSize"
-    viewBox="0 0 256 256"
-    xmlns="http://www.w3.org/2000/svg">
-    <path :d="path" />
-  </svg>
+  <component
+    :is="_icon"
+    :class="proportionClass" />
 </template>
 
-<script>
-import { ICONS, SIZES } from './util/iconVariables'
-import { hasOwnProp } from '~/utils'
-
-export default {
-  name: 'DpIcon',
-
-  props: {
-    icon: {
-      type: String,
-      required: true
-    },
-
-    size: {
-      type: String,
-      required: false,
-      default: 'medium',
-      validator: (prop) => Object.keys(SIZES).includes(prop)
-    }
+<script setup lang="ts">
+import {
+  computed,
+  markRaw,
+  PropType,
+  provide,
+  toRefs,
+  watchEffect
+} from 'vue'
+
+import {
+  iconComponents,
+  proportions,
+  SIZES
+} from './util/iconConfig'
+
+import {
+  IconName,
+  IconSize,
+  IconWeight
+} from '../../../types'
+
+import type { Component } from 'vue'
+
+const props = defineProps({
+  icon: {
+    type: String as PropType<IconName>,
+    required: true,
+    validator: (prop: IconName) => Object.keys(iconComponents).includes(prop)
   },
 
-  computed: {
-    pxSize () {
-      return SIZES[this.size]
-    },
-
-    path () {
-      if (hasOwnProp(ICONS[this.icon], 'path') && !!ICONS[this.icon].path) {
-        return ICONS[this.icon].path
-      } else {
-        console.warn(`DpIcon is called with an unsupported "icon" value.`)
-        return ICONS.warning.path
-      }
-    },
-
-    /**
-     * To enable authors to adapt spacing to different icon proportions, a selector
-     * is applied based on the width/height proportions of the path element.
-     */
-    proportionClass () {
-      return ICONS[this.icon]?.proportions || 'square'
-    }
+  /**
+   * Size of the icon
+   * Small: 16
+   * Medium: 20
+   * Large: 24
+   * Xlarge: 32
+   */
+  size: {
+    type: String as PropType<IconSize>,
+    required: false,
+    default: 'medium',
+    validator: (prop: IconSize) => ['small', 'medium', 'large', 'xlarge'].includes(prop)
```

Comment:
Could we drop the repeated values here by doing something like 
```suggestion
    validator: (prop: IconSize) => Object.keys(SIZES).includes(prop)
```

Reply:
Thanks, done in f5ad4b30416df009ba709e10493f8e2b79352ac8